scripts/build/autotools/CheckSSL.m4

```diff
@@ -0,0 +1,240 @@
+PHP_ARG_WITH([mongodb-ssl],
+             [whether to enable crypto and TLS],
+             [AS_HELP_STRING([--with-mongodb-ssl=@<:@auto/no/openssl/libressl/darwin@:>@],
+                             [Enable TLS connections and SCRAM-SHA-1 authentication [default=auto]])],
+             [auto],
+             [no])
+
+PHP_ARG_ENABLE([mongodb-crypto-system-profile],
+               [whether to use system crypto profile],
+               [AC_HELP_STRING([--enable-mongodb-crypto-system-profile],
+                               [Use system crypto profile (OpenSSL only) [default=no]])],
+               [no],
+               [no])
+
+PHP_ARG_WITH([openssl-dir],
+             [OpenSSL library path (deprecated)],
+             [AC_HELP_STRING([--with-openssl-dir=@<:@DIR@:>@],
+                             [OpenSSL library path (deprecated for pkg-config)])],
+             [yes],
+             [no])
+
+PHP_ARG_WITH([system-ciphers],
+             [whether to use system crypto profile (deprecated)],
+             AC_HELP_STRING([--enable-system-ciphers],
+                            [whether to use system crypto profile (deprecated for --enable-mongodb-crypto-system-profile) [default=no]]),
+             [no],
+             [no])
+
+AS_IF([test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "auto"],[
+  found_openssl="no"
+
+  PKG_CHECK_MODULES([PHP_MONGODB_SSL],[openssl],[
+    dnl PKG_CHECK_MODULES() doesn't check for headers, so confirm they exist (mainly for macOS)
+    old_CFLAGS="$CFLAGS"
+    CFLAGS="$PHP_MONGODB_SSL_CFLAGS"
+
+    AC_CHECK_HEADERS([openssl/bio.h openssl/ssl.h openssl/err.h openssl/crypto.h],
+                     [have_ssl_headers="yes"],
+                     [have_ssl_headers="no"])
+
+    if test "$have_ssl_headers" = "yes"; then
+      PHP_EVAL_INCLINE([$PHP_MONGODB_SSL_CFLAGS])
+      PHP_EVAL_LIBLINE([$PHP_MONGODB_SSL_LIBS],[MONGODB_SHARED_LIBADD])
+      PHP_MONGODB_SSL="openssl"
+      found_openssl="yes"
+    elif test "$PHP_MONGODB_SSL" = "openssl"; then
+      AC_MSG_ERROR([OpenSSL development headers could not be found])
+    fi
+
+    AC_CHECK_DECLS([ASN1_STRING_get0_data],
+                   [have_ASN1_STRING_get0_data="yes"],
+                   [have_ASN1_STRING_get0_data="no"],
+                   [[#include <openssl/asn1.h>]])
```

Comment:
Why are there two sets of [ ] here?

Reply:
Quoting https://lists.gnu.org/archive/html/bug-gnulib/2011-09/msg00172.html:

> # is a comment character in m4, which reads until end-of-line to end the comment. Which means all the [] after the comment character are treated as part of the comment, instead of as m4 [] quoting.

https://lists.gnu.org/archive/html/bug-gnulib/2011-09/msg00193.html discusses modifying `AC_CHECK_DECLS` to double-quote.

Note that we also used double-quoting in the [original check in `config.m4`](https://github.com/mongodb/mongo-php-driver/blob/e51a184717728e20f6f50280bf7dcd3156900951/config.m4#L309).